main.cpp

```diff
@@ -1,210 +1,69 @@
-	convertInfoIni(inDir, out, boundariesType, author, credit, license, cultureId, region, englishName);
-
-	AsterismOldLoader aLoader;
-	aLoader.load(inDir, cultureId);
-
-	ConstellationOldLoader cLoader;
-	cLoader.setBoundariesType(boundariesType.toStdString());
-	cLoader.load(inDir, outDir, nativeLocale);
-
-	NamesOldLoader nLoader;
-	nLoader.load(inDir, nativeLocale, convertUntranslatableNamesToNative);
-
-	std::cerr << "Starting emission of JSON...\n\n";
-	aLoader.dumpJSON(out);
-	cLoader.dumpJSON(out);
-	nLoader.dumpJSON(out);
-
-	auto str = std::move(out).str();
-	writeEnding(str);
-
-	if(!QDir().mkpath(outDir))
-	{
-		std::cerr << "Failed to create output directory\n";
-		return 1;
-	}
-	{
-		std::ofstream outFile((outDir+"/index.json").toStdString());
-		outFile << str;
-		outFile.flush();
-		if(!outFile)
-		{
-			std::cerr << "Failed to write index.json\n";
-			return 1;
-		}
-	}
-
-	DescriptionOldLoader dLoader;
-	license = convertLicense(license);
-	dLoader.load(inDir, poDir, cultureId, englishName, author, credit, license, cLoader, aLoader, nLoader,
-	             footnotesToRefs, genTranslatedMD);
-	dLoader.dump(outDir);
-
-	std::cerr << "--- NOTE ---\n";
-	if(region == "???")
-	{
-		std::cerr << "* Some JSON values can't be deduced from the old-format data. They have been"
-			     " marked by \"???\". Please replace them with something sensible.\n";
-	}
-	std::cerr << "* The key \"langs_use_native_names\" is omitted since it has no counterpart"
-	             " in the old format. If this sky culture needs it, please add it manually.\n";
-	std::cerr << "* The transformation of the description text is very basic, please check that"
-	             " it looks as it should. Pay special attention at References, Authors, and"
-	             " License sections, which may have been formulated in a suboptimal way.\n";
+        else if (arg == "--translated-md")
+            genTranslatedMD = true;
+        else if (arg == "--footnotes-to-references")
+            footnotesToRefs = true;
+        else if (arg == "--untrans-names-are-native")
+            convertUntranslatableNamesToNative = true;
+        else if (arg == "--native-locale")
+        {
+            static bool nextIsLocale = false;
+            nextIsLocale = !nextIsLocale;
+            if (nextIsLocale)
+                continue;
+            nativeLocale = arg;
+        }
+        else if (arg == "--help" || arg == "-h")
+        {
+            return usage(argv[0], 0);
+        }
+        else
+            return usage(argv[0], 1);
+    }
+    // invoke converter
+    auto result = SkyCultureConverter::convert(inDir, outDir, poDir, nativeLocale,
+                                              footnotesToRefs, genTranslatedMD,
+                                              convertUntranslatableNamesToNative);
+    std::cout << "SkyCultureConverter::\tConversion return-code: " << static_cast<int>(result) << "\n";
```

Comment:
Why print errors to stdout? They ought to go to stderr.

Reply:
Besides, if the converter prints errors by itself, you don't need to print this code there. And if it doesn't, you should give a user-friendly string that represents the error, not just some number.